library/psa_crypto.c

```diff
@@ -1185,9 +1205,20 @@ psa_status_t psa_get_key_attributes( mbedtls_svc_key_id_t key,
 
     psa_reset_key_attributes( attributes );
 
+#if defined(MBEDTLS_THREADING_C)
+    if( mbedtls_mutex_lock( &mbedtls_psa_slots_mutex ) != 0 )
```

Comment:
Why do you lock/unlock the mutex in PSA APIs and not just in `psa_get_and_lock_key_slot_in_memory` and `psa_get_empty_slot` ?

Reply:
I think letting several threads using the same key (read only) is a valid use case and if I am not mistaken this current approach does not allow that.

Reply:
When I think at the single mutex solution, the single mutex should only protect the reading and writing of some fields in the slots: the reading counter and a slot state (to be added, currently we check an `id` to determine if a slot is in used or not). Regarding slot states we need probably "not used", "in creation" (an empty slot has been selected and a key is currently being loaded into it), "usable", "in destruction". But this is only a sketch.

Reply:
This is a WIP, I simply wanted to catch any CI issues before diving deeper into the global lock approach. The idea is that, in the long run, we might want to use `psa_get_and_lock_key_slot_in_memory`, but with a flag to indicate if we want to read from or write to a slot, to implement a readers-writer lock.
I'm not sure what the real benefit of adding slot states would be (on a higher level), as this can be conveyed by the state of two mutexes used to implement the RW-lock, and might not be used by anything above it.

Reply:
Right, now I remember what was the problem: psa_copy_key locks two keyslots, so we can't go as low as `psa_get_and_lock_key_slot_in_memory` or `psa_get_empty_slot`.